src/coreclr/src/vm/genmeth.cpp

```diff
@@ -1676,6 +1676,7 @@ BOOL MethodDesc::SatisfiesMethodConstraints(TypeHandle thParent, BOOL fThrowIfNo
     //NB: according to the constructor's signature, thParent should be the declaring type,
     // but the code appears to admit derived types too.
     SigTypeContext typeContext(this,thParent);
+    InstantiationContext instContext(&typeContext, NULL);
```

Comment:
I am not sure whether substitutionChain is required here. The test does pass after passing in the InstantiationContext since the contraint gets the correct concrete type.

Reply:
No, it shouldn't be needed.

Reply:
Thanks!